pkg/cloudprovider/providers/azure/azure_util.go

```diff
@@ -365,3 +365,23 @@ func ExtractDiskData(diskData interface{}) (provisioningState string, diskState
 	}
 	return provisioningState, diskState, nil
 }
+
+// parseResourceGroupNameFromID parses the resource group name from a resource ID
+func parseResourceGroupNameFromID(resourceID string) (resourceGroupName string, err error) {
+	reg, err := regexp.Compile(`(?i)(.*?)/resourceGroups/(?P<rgname>\S+)/providers/(.*?)`)
```

Comment:
Consider using a `regexp.MustCompile` at module level (cf. `providerIDRE` on line 49 of this file).